### ISSUE TITLE
Extension enters error state on upgrade

### ISSUE THREAD

**Issue by nickurak** [mlutfy/hidetopbar#242]
Every time an upgrade for HideTopBar is installed from extensions,gnome.org, it stops working until I log out and log in again.

Most extensions work fine after an upgrade, and I was expecting the same here.

I've also tried the "Gnome Shell Extension Reloader" extension, which produces the same output.

The output comes from `journalctl  --user --grep 'hidetopbar@mathieu.bidon.ca'`:

```
-- Logs begin at Wed 2020-06-03 11:31:27 MDT, end at Fri 2020-07-03 12:44:07 MDT. --
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/extension.js
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/convenience.js
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/intellihide.js
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/prefs.js
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/ca/LC_MESSAGES/hi>
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/de/LC_MESSAGES/hi>
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/fr/LC_MESSAGES/hi>
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/it_IT/LC_MESSAGES>
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/pl/LC_MESSAGES/hi>
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/pt_BR/LC_MESSAGES>
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/zh_CN/LC_MESSAGES>
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:    creating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/schemas/
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/schemas/gschemas.compiled
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:   inflating: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/schemas/org.gnome.shell.>
Jul 03 10:00:25 bhtpc gnome-shell[1772183]:  extracting: /home/atrus/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/metadata.json
Jul 03 10:00:25 bhtpc gnome-shell[470780]: JS ERROR: Extension hidetopbar@mathieu.bidon.ca: Error: A different version was loaded previously. You need to log>
                                           loadExtension@resource:///org/gnome/shell/ui/extensionSystem.js:340:52
                                           callback@resource:///org/gnome/shell/ui/extensionDownloader.js:220:39
                                           gotExtensionZipFile/<@resource:///org/gnome/shell/ui/extensionDownloader.js:106:13

```

**Comment by nickurak** [comment]
In case compatability with other extensions is an issue, here's the state of other extensions.

`for E in $(gnome-extensions list); do echo -e "$(gnome-extensions info "$E" | grep State | awk '{print $2}')\t$E"; done`

```
ENABLED	Slack_Integration@rickycezar.gmail.com
ENABLED	appindicatorsupport@rgcjonas.gmail.com
ENABLED	audio-switcher@AndresCidoncha
ENABLED	desk-changer@eric.gach.gmail.com
ENABLED	pixel-saver@deadalnix.me
ENABLED	caffeine@patapon.info
ENABLED	native-window-placement@gnome-shell-extensions.gcampax.github.com
ENABLED	drive-menu@gnome-shell-extensions.gcampax.github.com
ENABLED	lockkeys@vaina.lt
ENABLED	audio-output-switcher@anduchs
ENABLED	extension-reloader@nls1729
INITIALIZED	apps-menu@gnome-shell-extensions.gcampax.github.com
INITIALIZED	background-logo@fedorahosted.org
INITIALIZED	horizontal-workspaces@gnome-shell-extensions.gcampax.github.com
INITIALIZED	launch-new-instance@gnome-shell-extensions.gcampax.github.com
INITIALIZED	places-menu@gnome-shell-extensions.gcampax.github.com
INITIALIZED	window-list@gnome-shell-extensions.gcampax.github.com
INITIALIZED	user-theme@gnome-shell-extensions.gcampax.github.com
INITIALIZED	gamemode@christian.kellner.me
ERROR	hidetopbar@mathieu.bidon.ca
```

**Comment by SerhiyRomanov** [comment]
I'm faced with the same problem - after updated it doesn't work. 
My env - Ubuntu 20.04, GNOME Shell 3.36.3

**Comment by coldtobi** [comment]
> My env - Ubuntu 20.04, GNOME Shell 3.36.3

Can you try the Debian package: 
`sudo apt install gnome-shell-extension-autohidetopbar`

**Comment by SerhiyRomanov** [comment]
@coldtobi I've tried it and it works without any issues) Thanks for help!

**Comment by magluf** [comment]
I got this error with the most recent update.

Fedora 32, GNOME Shell 3.36.4.

![image](https://user-images.githubusercontent.com/32460935/90089609-2b9e0d80-dcf8-11ea-8443-751fa1d9b9a4.png)

```
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/extension.js
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/convenience.js
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/intellihide.js
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/prefs.js
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/ca/LC_MESSAGES/hidetopbar.mo
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/de/LC_MESSAGES/hidetopbar.mo
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/fr/LC_MESSAGES/hidetopbar.mo
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/it_IT/LC_MESSAGES/hidetopbar.mo
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/pl/LC_MESSAGES/hidetopbar.mo
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/pt_BR/LC_MESSAGES/pt_BR.mo
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/locale/zh_CN/LC_MESSAGES/hidetopbar.mo
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:    creating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/schemas/
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/schemas/gschemas.compiled
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:   inflating: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/schemas/org.gnome.shell.extensions.hidetopbar.gschema.>
Aug 12 22:34:24 localhost.localdomain gnome-shell[16912]:  extracting: /home/magluf/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/metadata.json
Aug 12 22:34:24 localhost.localdomain gnome-shell[2033]: JS ERROR: Extension hidetopbar@mathieu.bidon.ca: Error: A different version was loaded previously. You need to log out for changes to take effect.
                                                         loadExtension@resource:///org/gnome/shell/ui/extensionSystem.js:340:52
                                                         callback@resource:///org/gnome/shell/ui/extensionDownloader.js:220:39
                                                         gotExtensionZipFile/<@resource:///org/gnome/shell/ui/extensionDownloader.js:106:13
```

**Comment by coldtobi** [comment]
```
Aug 12 22:34:24 localhost.localdomain gnome-shell[2033]: JS ERROR: Extension hidetopbar@mathieu.bidon.ca: Error: A different version was loaded previously. You need to log out for
+changes to take effect.
```
Did you log out and log in?

**Comment by nickurak** [comment]
> Did you log out and log in again?

The point of this issue is you shouldn't have to log out and log in.

**Comment by coldtobi** [comment]
AFAIK you need to, at least on Wayland. (gnome shell tells me there it cannot reload.)

**Comment by nickurak** [comment]
Right, but you shouldn't have to reload Gnome-shell either.

Most extensions just upgrade in place, no restarting of anything.

**Comment by tuxor1337** [comment]
Other extensions get the same error message on update: https://github.com/corecoding/Vitals/issues/165 https://github.com/mzur/gnome-shell-wsmatrix/issues/118 https://github.com/ai/autohide-battery/issues/13

@nickurak "You shouldn't have to reload gnome-shell". Actually, reloading is enforced by gnome-shell. It's impossible that any other extension is able to upgrade in place: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1248

To summarize, manually updating extensions without restarting GNOME does never work. The manual update feature on extensions.gnome.org should simply not exist. It is broken and nobody needs it, because extension updates are performed automatically when logging of and on again.
